### ISSUE TITLE
Separate out `Server` header's functional version number from binary build number

### ISSUE THREAD

**Issue by zgramana** [couchbase/sync_gateway#1209]

**Comment by snej** [comment]
The HTTP/1.1 spec has [rules for the Server header](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.38). Basically it needs to be one or more _name_/_version_ pairs separated by spaces.

It could be something like:

```
Couchbase Sync Gateway/1.3 commit/#29AE23F0
```

As long as it starts with `Couchbase Sync Gateway/` followed by a nonzero decimal digit, it'll be compatible with existing versions of Couchbase Lite.

**Comment by snej** [comment]
Bad news: our current Server header value is invalid according to spec because the product name (`Couchbase Sync Gateway`) contains spaces. The [HTTP/1.1 grammar](http://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2.2) says that a `product` token can't contain whitespace, or various other punctuation characters. We should change it to `CouchbaseSyncGateway` or `Couchbase-Sync-Gateway` or something like that.

Of course, if we change this it will confuse existing versions of Couchbase Lite; they'll stop recognizing the server as Sync Gateway and won't use replication optimizations like bulk_get. We probably don't want to do that :(

**Comment by snej** [comment]
I've updated the header value for unofficial builds; it now looks like

```
Server: Couchbase Sync Gateway/1.2 branch/fix/server_header commit/5bfcf79+CHANGES
```
